cli/schoolAdmin_parentWeeklyEmailSummary.php

```diff
@@ -17,340 +17,281 @@
+    echo __('School is not open, so no emails will be sent.');
+    return;
+}
+
+if ($_SESSION[$guid]['organisationEmail'] == '') {
+    echo __('This script cannot be run, as no school email address has been set.');
+    return;
 }
-else {
-    //Check that one of the days in question is a school day
-    $isSchoolOpen = false;
-    for ($i = 0; $i < 7; ++$i) {
-        if (isSchoolOpen($guid, date('Y-m-d', strtotime("-$i day")), $connection2, true) == true) {
-            $isSchoolOpen = true;
+
+$gibbonSchoolYearID = $gibbon->session->get('gibbonSchoolYearID');
+$parentWeeklyEmailSummaryIncludeBehaviour = getSettingByScope($connection2, 'School Admin', 'parentWeeklyEmailSummaryIncludeBehaviour');
+$parentWeeklyEmailSummaryIncludeMarkbook = getSettingByScope($connection2, 'School Admin', 'parentWeeklyEmailSummaryIncludeMarkbook');
+$sendReport = ['emailSent' => 0, 'emailFailed' => 0, 'emailErrors' => ''];
+
+// Prep for email sending later
+$mail = $container->get(Mailer::class);
+$mail->SMTPKeepAlive = true;
+
+$familyGateway = $container->get(FamilyGateway::class);
+$rollGroupGateway = $container->get(RollGroupGateway::class);
+$plannerEntryGateway = $container->get(PlannerEntryGateway::class);
+$emailSummaryGateway = $container->get(PlannerParentWeeklyEmailSummaryGateway::class);
+$view = $container->get(View::class);
+
+// Get all student data grouped by family
+$families = $familyGateway->selectFamiliesWithActiveStudents($gibbonSchoolYearID)->fetchGrouped();
+
+foreach ($families as $gibbonFamilyID => $students) {
+    // Get the adults in this family and filter by email settings
+    $familyAdults = $familyGateway->selectAdultsByFamily($gibbonFamilyID, true)->fetchAll();
+    $familyAdults = array_filter($familyAdults, function ($parent) {
+        return $parent['contactEmail'] == 'Y';
+    });
+
+    if (empty($familyAdults)) continue;
+
+    foreach ($students as $student) {
+
+        // HOMEWORK
+
+        $criteria = $plannerEntryGateway->newQueryCriteria(true)
+            ->sortBy(['date', 'timeStart'], 'DESC')
+            ->filterBy('weekly:Y')
+            ->filterBy('viewableParents:Y')
+            ->fromPOST();
+
+        $allHomework = $plannerEntryGateway->queryHomeworkByPerson($criteria, $gibbonSchoolYearID, $student['gibbonPersonID']);
+
+        $tracker = $plannerEntryGateway->selectHomeworkTrackerByStudent($gibbonSchoolYearID, $student['gibbonPersonID'])->fetchGroupedUnique();
+        $allHomework->joinColumn('gibbonPlannerEntryID', 'tracker', $tracker);
+
+        $submissions = $plannerEntryGateway->selectHomeworkSubmissionsByStudent($gibbonSchoolYearID, $student['gibbonPersonID'])->fetchGrouped();
+        $allHomework->joinColumn('gibbonPlannerEntryID', 'submissions', $submissions);
+
+
+        $dataHomework = ['gibbonPersonID' => $student['gibbonPersonID'], 'gibbonSchoolYearID' => $_SESSION[$guid]['gibbonSchoolYearID'], 'today' => date('Y-m-d'), 'lastWeek' => date('Y-m-d', strtotime('-1 week'))];
```

Comment:
Does this not do the same as line 98 above?

Reply:
Yep, dead code, fixed!